examples/ChatQnA/ChatQnA_Guide.rst

```diff
@@ -142,20 +132,85 @@ See :doc:`/GenAIExamples/supported_examples`
 Architecture
 ************
 
-TODO: Includes microservice level graphics.
+The ChatQnA architecture is displayed below:
+![alt text](image.png)
+
+.. figure:: /GenAIExamples/ChatQnA/assets/img/chatqna_flow_chart.png
+   :alt: ChatQnA Architecture Diagram
+
+.. code-block:: python
+
+graph LR
+    subgraph ChatQnA-MegaService["ChatQnA-MegaService"]
+        direction LR
+        EM([Embedding 'LangChain TEI' <br>6000])
+        RET([Retrieval 'LangChain Redis'<br>7000])
+        RER([Rerank 'TEI'<br>8000])
+       LLM([LLM 'text-generation TGI'<br>9000])
+    end
+
+    direction TB
+    TEI_EM{{TEI embedding service<br>8090}}
+    VDB{{Vector DB<br>8001}}
+    %% Vector DB interaction
+    TEI_EM -.->|d|VDB
+
+    DP([OPEA Data Preparation<br>6007])
+    LLM_gen{{TGI/vLLM/ollama Service}}
+
+    direction TB
+    RER([OPEA Reranking<br>8000])
+    TEI_RER{{TEI Reranking service<br>8808}}
+
+    subgraph User Interface
+        direction TB
+        a[User Input Query]
+        Ingest[Ingest data]
+        UI[UI server<br>Port: 5173]
+    end
+
+    subgraph ChatQnA GateWay
+        direction LR
+        GW[ChatQnA GateWay<br>Port: 8888]
+    end
+
+    %% Data Preparation flow
+    %% Ingest data flow
+    direction LR
+    Ingest[Ingest data] -->|a| UI
+    UI -->|b| DP
+    DP -.->|c| TEI_EM
+
+    %% Questions interaction
+    direction LR
+    a[User Input Query] -->|1| UI
+    UI -->|2| GW
+    GW ==>|3| ChatQnA-MegaService
+    EM ==>|4| RET
+    RET ==>|5| RER
+    RER ==>|6| LLM
+
+
+    %% Embedding service flow
+    direction TB
+    EM -.->|3'| TEI_EM
+    RET -.->|4'| TEI_EM
+    RER -.->|5'| TEI_RER
+    LLM -.->|6'| LLM_gen
+
+    subgraph Legend
+        X([Micsrservice])
+        Y{{Service from industry peers}}
+        Z[Gateway]
+    end
 
-TODO: Need to include the architecture with microservices. Like the ones
-Xigui/Chun made and explain in a paragraph or 2 on the highlights of the arch
-including Gateway, UI, mega service, how models are deployed and how the
-microservices use the deployment service. The architecture can be laid out as
-general as possible, maybe  calling out “for e.g” on variable pieces. Will also
-be good to include a line or 2 on what the overall use case is. For e.g. This
-chatqna is setup to assist in answering question on OPEA. The microservices are
-set up with RAG and LLM pipeline to query on OPEA PDF documents
 
 Microservice Outline and Diagram
 ================================
 
+OPEA projects typcially consist of a collection of microservices to create a megaservice, all facilitated by a gateway. A microservice is a component designed to perform a specific function or task. Microservices are building blocks, offering the fundamental services. Microservices promote the modularity, flexibility, and scalability of the system. A megaservice is a higher-level architectural construct composed of one or more microservices, providing the capability to assemble end-to-end applications.
```

Comment:
It's a good practice to keep line length around 80 characters so you don't have these really long lines in Git to deal with.